sys/socket.c

```diff
@@ -435,3 +436,122 @@ int socketpair(int domain, int type, int protocol, int socket_vector[2])
 
 	return 0;
 }
+
+
+int getifaddrs(struct ifaddrs **ifap)
+{
+	struct ifaddrs *ifa;
+	msg_t msg = { 0 };
+	sockport_resp_t *smo = (void *)msg.o.raw;
+	size_t bufsz, namesz, addrsz;
+	char *p;
+
+	msg.type = sockmGetIfAddrs;
+	msg.o.size = 256;
+
+	for (;;) {
+		void *buf = realloc(msg.o.data, msg.o.size);
+		if (!buf) {
+			realloc(msg.o.data, 0);
```

Comment:
```suggestion
			free(msg.o.data);
```
In `libphoenix` this functionality is probably not implement.